README.md

```diff
@@ -76,11 +67,23 @@ A valid (read: that follows this interface) IPLD format implementation the follo
 
 `callback` must have the signature `function (err, dagNode)`, where `err` is an Error if the function fails and `dagNode` is the dagNode that got deserialized in the process.
 
-#### `util.cid(dagNode, callback)`
+#### `util.cid(binaryBlob[, options], callback)`
+
+> get the CID of the serialized dagNode
+
+Options include:
+  - version - the CID version to be used (defaults to 1)
+  - hashAlg - the hash algorithm to be used (default to the one set by the format)
+
+`callback` must have the signature `function (err, cid)`, where `err` is an Error if the function fails and `cid` is a CID instance of the serialized dagNode.
 
-> get the CID of the dagNode
+#### `util.toJSON(dagNode, callback)`
 
-`callback` must have the signature `function (err, cid)`, where `err` is an Error if the function fails and `cid` is a CID instance of the dagNode.
+> return a JSON representation of the dagNode
```

Comment:
Is this turning the DAG node into JSON or a JavaScript object? I ask because, as it stands, we haven't defined a way to represent all IPLD objects in JSON (binary data and types are problematic).

Reply:
My plan is to have JSON so other languages like Go can also implement it. But it's a good point that we need to come up with a way to deal with binary data.

Based on @kumavis comment, I'll probably start with using the output of `tree()` to create JSON out of it.

Reply:
> My plan is to have JSON so other languages like Go can also implement it. 

I'd rather use CBOR as the interchange format. Really, we should have a general-purpose system for converting between one format and another.

> But it's a good point that we need to come up with a way to deal with binary data.

The consensus here was to *not* do this and to say that DagJSON is a lesser format that doesn't support binary. My original suggestion was to use `{"/bytes": "multibase encoded data"}` in the same manner as `{"/": Cid}`.

Reply:
I don't want to use JSON as interchange format as you can't represent everything with JSON. I'd rather use it as "let's see what's in there" and "easy to index" format.

Multibase encoding or even just not returning the data if there isn't a nice JSON representation is fine with me.

Reply:
I see. I'm fine with that (go has it as well) although I'm not sure if it really belongs on a language independent spec. However, given that this is, effectively, a JS spec, meh?

Reply:
We need to pick one. Either:

1. JSON is a universal, human-readable exchange format. In this case, it needs to support binary.
2. `toJSON` is for debugging. In this case, it doesn't belong in a language independent spec.

I'm going to try turning off my email to see if I can spec something out.